src/internal/connector/data_collections.go

```diff
@@ -229,3 +241,109 @@ func (gc *GraphConnector) OneDriveDataCollections(
 
 	return collections, errs
 }
+
+// ---------------------------------------------------------------------------
+// SharePoint
+// ---------------------------------------------------------------------------
+
+// createSharePointCollections - utility function that retrieves M365
+// IDs through Microsoft Graph API. The selectors.SharePointScope
+// determines the type of collections that are retrieved.
+func (gc *GraphConnector) createSharePointCollections(
+	ctx context.Context,
+	scope selectors.SharePointScope,
+) ([]*sharepoint.Collection, error) {
+	var (
+		errs  *multierror.Error
+		sites = scope.Get(selectors.SharePointSite)
+		colls = make([]*sharepoint.Collection, 0)
+	)
+
+	// Create collection of ExchangeDataCollection
+	for _, site := range sites {
+		collections := make(map[string]*sharepoint.Collection)
+
+		qp := graph.QueryParams{
+			// TODO: Resource owner, not user/site.
+			User: site,
+			// TODO: generic scope handling in query params.
+			// - or, break scope out of QP.
+			// Scope:       scope,
+			FailFast:    gc.failFast,
+			Credentials: gc.credentials,
+		}
+
+		itemCategory := qp.Scope.Category().PathType()
+
+		foldersComplete, closer := observe.MessageWithCompletion(fmt.Sprintf("∙ %s - %s:", itemCategory.String(), site))
+		defer closer()
+		defer close(foldersComplete)
+
+		resolver, err := exchange.PopulateExchangeContainerResolver(
```

Comment:
Unsure if this is the correct set-up. List and Drive items are different when it comes to paths. Need to ensure that the resolvers are also different for these two items.